src/App.jsx

```diff
@@ -14,45 +17,112 @@ export const goodsFromServer = [
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+const getSortedGoods = (goods, sortField, isReversed) => {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const GoodList = ({ goods }) => (
+  <ul>
+    {goods.map(
+      good => (
+        <li
+          data-cy="Good"
+          key={good}
+        >
+          {good}
+        </li>
+      ),
+    )}
+  </ul>
 );
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversedList, setReversedList] = useState(false);
+  const visibleGoods = getSortedGoods(goodsFromServer, sortField, reversedList);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-info',
+              { 'is-light': sortField !== SORT_FIELD_ALPHABET },
+            )
+          }
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-success',
+              { 'is-light': sortField !== SORT_FIELD_LENGTH },
+            )
+          }
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button',
+              'is-warning',
+              { 'is-light': !reversedList },
+            )
+          }
+          onClick={() => setReversedList(!reversedList)}
+        >
+          Reverse
+        </button>
+
+        {(sortField.length > 0 || reversedList) && (
+          <button
+            type="button"
+            className={cn('button', 'is-danger', 'is-light')}
+            onClick={() => {
+              setSortField('');
+              setReversedList(false);
+            }}
```

Comment:
It's better to create separate functions to handle events. This way, your code will be easier to maintain and change
```suggestion
            onClick={onReset}
```